dcm4che-core/src/main/java/org/dcm4che3/io/DicomInputStream.java

```diff
@@ -548,11 +548,30 @@ public Attributes readDatasetUntilPixelData() throws IOException {
         return readDataset(-1, o -> o.tag == Tag.PixelData);
     }
 
+    /**
+     * @deprecated Use one of the other {@link #readDataset()} methods instead. If you want to
+     * specify a length limit, you may supply a {@link LimitedInputStream} or use
+     * {@link #createWithLimit} or {@link #createWithLimitFromFileLength}.
+     */
     @Deprecated
     public Attributes readDataset(int len, int stopTag) throws IOException {
         return readDataset(len, tagEqualOrGreater(stopTag));
     }
 
+    public Attributes readDataset(int stopTag) throws IOException {
```

Comment:
Not sure why the above method was already deprecated before. I actually think using a stopTag can be quite useful, therefore I have added this method here.